### PR TITLE
Delete experimental radix-ui based DropdownMenu component

### PR DESCRIPTION
**Pull request by ciampo**
Requires #55625

## What?

Remove the experimental, temporary version of `DropdownMenu` based off Radix UI

## Why?

The Radix UI version is being removed in favour of the ariakit-based version (see #50459 for more details)

## How?

- Delete all of the code related to the radix UI -based `DropdownMenuV2`
- Remove `radix-ui` as a dependency of the project
- Rename the ariakit based `DropdownMenuV2Ariakit` component to `DropdownMenuV2` (including folder structure).

**I recommend reviewing commit-by-commit.**

## Testing Instructions

- Make sure the project builds correctly
- Make sure that there are no traces left of Radix UI and the Radix UI-based DropdownMenu
- Make sure that the Storybook examples for the DropdownMenuV2 work as expected (ariakit implementation)
- Make sure that the DropdownMenus used in the dataviews package work as expected (see #55625 for more context)

### CODE REVIEW

File: packages/dataviews/src/view-actions.js [hunk 1]
```diff
@@ -15,11 +15,11 @@ import { VIEW_LAYOUTS, LAYOUT_TABLE, SORTING_DIRECTIONS } from './constants';
...
-	DropdownMenuV2Ariakit: DropdownMenu,
-	DropdownMenuGroupV2Ariakit: DropdownMenuGroup,
-	DropdownMenuItemV2Ariakit: DropdownMenuItem,
-	DropdownMenuCheckboxItemV2Ariakit: DropdownMenuCheckboxItem,
-	DropdownMenuItemLabelV2Ariakit: DropdownMenuItemLabel,
+	DropdownMenuV2: DropdownMenu,
+	DropdownMenuGroupV2: DropdownMenuGroup,
+	DropdownMenuItemV2: DropdownMenuItem,
+	DropdownMenuCheckboxItemV2: DropdownMenuCheckboxItem,
+	DropdownMenuItemLabelV2: DropdownMenuItemLabel,
```
Comment (mirka) [inline packages/dataviews/src/view-actions.js:22]: Aside: Not sure if I talked to you about this, but I'm thinking we should switch to dot notation exports (`DropdownMenuV2.Group`, etc) for better ergonomics all around. It's really cumbersome with all the unlocking and aliasing.

**Review by mirka — APPROVED** [review]
Do you still want to keep the [Radix section in the Contributing Guide](https://github.com/WordPress/gutenberg/blob/0080b752d8d980c8f06bb0ab27729f6a74f2e367/packages/components/CONTRIBUTING.md?plain=1#L596-L603)?

Otherwise, looks good 👍

File: packages/dataviews/src/view-actions.js (hunk 1 above)
Comment (ciampo) [reply packages/dataviews/src/view-actions.js:22]: Yes, I'd like to switch to dot notation too. Maybe call the "parent" component `DropdownMenu.Root` too. Let's discuss / address in a separate PR!

**Comment by ciampo** [comment]
> Do you still want to keep the [Radix section in the Contributing Guide](https://github.com/WordPress/gutenberg/blob/0080b752d8d980c8f06bb0ab27729f6a74f2e367/packages/components/CONTRIBUTING.md?plain=1#L596-L603)?

Makes sense, opened https://github.com/WordPress/gutenberg/pull/57336

**Pull request merged**